recipes/openvdb/8.0.1/conanfile.py

```diff
@@ -0,0 +1,232 @@
+        "fPIC": [True, False],
+        "with_blosc": [True, False],
+        "with_zlib": [True, False],
+        "with_log4cplus": [True, False],
+        "with_exr": [True, False],
+        "simd": ["None", "SSE42", "AVX"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_blosc": True,
+        "with_zlib": True,
+        "with_log4cplus": False,
+        "with_exr": False,
+        "simd": "None",
+    }
+
+    _cmake = None
+
+    _compilers_min_version = {
+        "msvc": "19.10",
+        "Visual Studio": "15",  # Should we check toolset?
+        "gcc": "6.3.1",
+        "clang": "3.8",
+        "apple-clang": "3.8",
+        "intel": "17",
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _check_compilier_version(self):
+        compiler = str(self.settings.compiler)
+        version = str(self.settings.compiler.version)
+        if version < self._compilers_min_version[compiler]:
+            raise ConanInvalidConfiguration("%s requires a %s version greater than %s" % (self.name, compiler, self._compilers_min_version[compiler]))
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
+        if self.settings.arch not in ("x86", "x86_64"):
+            if self.options.simd != "None":
+                raise ConanInvalidConfiguration("Only intel architectures support SSE4 or AVX.")
+        self._check_compilier_version()
+
+    def requirements(self):
+        self.requires("boost/1.75.0")
+        self.requires("tbb/2020.3")
+        self.requires("openexr/2.5.5")  # required for IlmBase::Half
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
+        if self.options.with_exr:
+            # Not necessary now. Required for IlmBase::IlmImf
+            self.requires("openexr/2.5.5]")
+        if self.options.with_blosc:
+            self.requires("c-blosc/1.20.1")
+        if self.options.with_log4cplus:
+            self.requires("log4cplus/2.0.5")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _patch_sources(self):
+        # Remove FindXXX files from OpenVDB. Let Conan do the job
+        tools.remove_files_by_mask(os.path.join(self._source_subfolder, "cmake"), "Find*")
+        with open("FindBlosc.cmake", "w") as f:
+            f.write(
+                """find_package(c-blosc)
+if(c-blosc_FOUND)
+    add_library(blosc INTERFACE)
+    target_link_libraries(blosc INTERFACE c-blosc::c-blosc)
+    add_library(Blosc::blosc ALIAS blosc)
+endif()
+"""
+            )
```

Comment:
```suggestion
        with open("FindBlosc.cmake", "w") as f:
            f.write(
                """find_package(c-blosc)
if(c-blosc_FOUND)
    add_library(blosc INTERFACE)
    target_link_libraries(blosc INTERFACE c-blosc::c-blosc)
    add_library(Blosc::blosc ALIAS blosc)
endif()
"""
            )
```

Did you try [cmake_find_package](https://docs.conan.io/en/latest/reference/generators/cmake_find_package.html) generator instead?

Reply:
Yes, I did, but package names are different so instead of patching the CMakelists I wrapped the conan libraries with the expected name as @madebr suggested [here](https://github.com/conan-io/conan-center-index/pull/5257#discussion_r615432593)